earth2mip/initial_conditions/ifs.py

```diff
@@ -49,75 +56,109 @@ def _get_channel(c: str, **kwargs) -> xarray.DataArray:
+
+    # split control forecast and perturbed forecasts 
+    dataset_pf = [ds for ds in dataset_0h if 0 not in ds['number']]
+    dataset_cf = [ds for ds in dataset_0h if 0 in ds['number']]
+
+    if ensemble_member == 0:
+        channel_data = [
+            _get_channel(
+                c,
+                u10m=[ds['u10'] for ds in dataset_cf if 'u10' in ds.data_vars][0],
+                v10m=[ds['v10'] for ds in dataset_cf if 'v10' in ds.data_vars][0],
+                u100m=[ds['u100'] for ds in dataset_cf if 'u100' in ds.data_vars][0],
+                v100m=[ds['v100'] for ds in dataset_cf if 'v100' in ds.data_vars][0],
+                sp=[ds['sp']  for ds in dataset_cf if 'sp' in ds.data_vars][0],
+                t2m=[ds['t2m'] for ds in dataset_cf if 't2m' in ds.data_vars][0],
+                msl=[ds['msl'] for ds in dataset_cf if 'msl' in ds.data_vars][0],
+                tcwv=[ds['tcwv'] for ds in dataset_cf if 'tcwv' in ds.data_vars][0],
+                t=[ds['t'] for ds in dataset_cf if 't' in ds.data_vars][0],
+                u=[ds['u'] for ds in dataset_cf if 'u' in ds.data_vars][0],
+                v=[ds['v'] for ds in dataset_cf if 'v' in ds.data_vars][0],
+                r=[ds['r'] for ds in dataset_cf if 'r' in ds.data_vars][0],
+                z=[ds['gh'] for ds in dataset_cf if 'gh' in ds.data_vars][0] * 9.81,
+            )
+            for c in channels
+        ]
+    else: 
+        channel_data = [
+            _get_channel(
+                c,
+                u10m=[ds['u10'] for ds in dataset_pf if 'u10' in ds.data_vars][0].sel(number=ensemble_member),
+                v10m=[ds['v10'] for ds in dataset_pf if 'v10' in ds.data_vars][0].sel(number=ensemble_member),
+                u100m=[ds['u100'] for ds in dataset_pf if 'u100' in ds.data_vars][0].sel(number=ensemble_member),
+                v100m=[ds['v100'] for ds in dataset_pf if 'v100' in ds.data_vars][0].sel(number=ensemble_member),
+                sp=[ds['sp']  for ds in dataset_pf if 'sp' in ds.data_vars][0].sel(number=ensemble_member),
+                t2m=[ds['t2m'] for ds in dataset_pf if 't2m' in ds.data_vars][0].sel(number=ensemble_member),
+                msl=[ds['msl'] for ds in dataset_pf if 'msl' in ds.data_vars][0].sel(number=ensemble_member),
+                tcwv=[ds['tcwv'] for ds in dataset_pf if 'tcwv' in ds.data_vars][0].sel(number=ensemble_member),
+                t=[ds['t'] for ds in dataset_pf if 't' in ds.data_vars][0].sel(number=ensemble_member),
+                u=[ds['u'] for ds in dataset_pf if 'u' in ds.data_vars][0].sel(number=ensemble_member),
+                v=[ds['v'] for ds in dataset_pf if 'v' in ds.data_vars][0].sel(number=ensemble_member),
+                r=[ds['r'] for ds in dataset_pf if 'r' in ds.data_vars][0].sel(number=ensemble_member),
+                z=[ds['gh'] for ds in dataset_pf if 'gh' in ds.data_vars][0].sel(number=ensemble_member) * 9.81,
+            )
+            for c in channels
+        ]
```

Comment:
(I left this comment in my last review but must have forgot to hit send)

This is not necessary for the MR and we can go ahead and merge, but I thought of this:
```python
channel_vars =  ['u10', 'v10', 'u100', 'v100', 'sp', 't2m', 'msl', 'tcwv', 't', 'u', 'v', 'r']

if ensemble_member == 0:
    dset = [ds for ds in dataset_0h if 0 in ds['number']]
    def _subset(varname):
        return [ds[varname] for ds in dset if varname in ds.data_vars][0]
else:
    dset = [ds for ds in dataset_0h if 0 not in ds['number']]
    def _subset(varname):
        return [ds[varname] for ds in dset if varname in ds.data_vars][0].sel(number=ensemble_member)

channel_data = [
    _get_channel(
        c,
        **{v: _subset(v) for v in channel_vars},
        z=_subset('gh') * 9.81,
    )
    for c in channels
]
```

Reply:
good suggestion to clean this bit up - the above doesn't quite work because some of the channel variable names have to be reassigned, so they need to be treated separately as `z` is above.